docs/pages/access-controls/sso/github-sso.mdx

```diff
@@ -32,86 +32,110 @@ https://PROXY_ADDRESS/v1/webapi/github/
+        - editor
+      team: GITHUB-TEAM
+version: v3
 ```
 
-</TabItem>
-<TabItem label="Enterprise / Cloud" scope={["enterprise", "cloud"]}>
+<Details title="Self-hosting GitHub Enterprise?" scope={["enterprise", "cloud"]} opened>
+
+For self-hosted GitHub Enterprise servers, add `spec.endpoint_url` pointing to your server instance:
 
 ```yaml
-kind: github
-version: v3
-metadata:
-  # Connector name that will be used with `tsh --auth=github login`
-  name: github
+...
 spec:
-  # Client ID of your GitHub OAuth App
-  client_id: <client-id>
-  # Client secret of your GitHub OAuth App
-  client_secret: <client-secret>
-  # Connector display name that will be shown on the Web UI login screen
-  display: GitHub
-  # URL of your GitHub Enterprise Server instance (if applicable)
+  ...
   endpoint_url: https://<github-enterprise-server-address>
-  # Callback URL that will be called after successful authentication
-  redirect_url: https://<proxy-address>/v1/webapi/github/callback
-  # Mapping of org/team memberships onto allowed roles
-  teams_to_roles:
-    - organization: octocats # GitHub organization name
-      team: admins # GitHub team name within that organization
-      # Maps octocats/admins to the "access" Teleport role
-      roles:
-        - access
+  ...
 ```
 
-</TabItem>
-</Tabs>
-
-The values of `client_id`, `client_secret`, and `redirect_url` come from the
-GitHub OAuth App you created earlier.
+</Details>
 
-Teleport will request only the `read:org` OAuth scope. Read more about OAuth scopes in GitHub's documentation:
+You can optionally test the connector configuration before applying it to your cluster:
 
-[GitHub OAuth scopes](https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/)
+```code
+$ cat github.yaml | tctl sso test
+If browser window does not open automatically, open it by clicking on the link:
+ http://127.0.0.1:52690/35714f6b-...
+Success! Logged in as: alice
+--------------------------------------------------------------------------------
+Authentication details:
+   roles:
+   - access
+   - editor
+   traits:
+     github_teams:
+     - admins
+     kubernetes_groups: null
+     kubernetes_users: null
+     logins:
+     - alice
+   username: alice
+--------------------------------------------------------------------------------
+[GitHub] Received claims:
+organization_to_teams:
+  Octocats:
+  - admins
+teams:
+- admins
+username: alice
+
+--------------------------------------------------------------------------------
+[GitHub] Connector team to roles mapping:
+- organization: Octocats
+  roles:
+  - access
+  - editor
+  team: admins
+
+--------------------------------------------------------------------------------
+For more details repeat the command with --debug flag.
+```
 
 Finally, create the connector using `tctl`:
 
 ```code
 $ tctl create github.yaml
-# authentication connector "github" has been created
+authentication connector "github" has been created
```

Comment:
On the second attempt, this will fail as the resource will already exist by that point. I wonder if we should mention the `-f` option here to solve that?

Reply:
Adding `-f`, which is standard (I forgot) in docs for this very reason. Regarding alternate flows, I think at some point we have to trust the reader to know this sort of bash piping syntax and use whatever they're comfortable with.

Reply:
Resolved in 9dafbcfabff402479db7a487e4f58a52ea44b446